include/full_coverage_path_planner/full_coverage_path_planner.hpp

```diff
@@ -100,8 +91,16 @@ class FullCoveragePathPlanner
                  float toolRadius,
                  geometry_msgs::msg::PoseStamped const& realStart,
                  Point_t& scaledStart);
-  ros::Publisher plan_pub_;
-  ros::ServiceClient cpp_grid_client_;
+
+  /**
+   * @brief Create Quaternion from Yaw
+   * @param yaw orientation
+   * @return Quaternion with desired yaw orientation
+   */
+  auto createQuaternionMsgFromYaw(double yaw);
```

Comment:
These very small helper functions you may also simply write out fully in the header:

```c++
auto createQuaternionMsgFromYaw(double yaw)
{
  tf2::Quaternion q;
  q.setRPY(0, 0, yaw);
  return tf2::toMsg(q);
}
```